apps/dashboard/app/(app)/logs/components/charts/hooks.ts

```diff
@@ -0,0 +1,65 @@
+import { trpc } from "@/lib/trpc/client";
+import type { LogsTimeseriesDataPoint } from "@unkey/clickhouse/src/logs";
+import { addMinutes, format } from "date-fns";
+import { useMemo } from "react";
+import { useLogSearchParams } from "../../query-state";
+import { type TimeseriesGranularity, getTimeseriesGranularity } from "../../utils";
+
+const roundToSecond = (timestamp: number) => Math.floor(timestamp / 1000) * 1000;
+
+const formatTimestamp = (value: string | number, granularity: TimeseriesGranularity) => {
+  const date = new Date(value);
+  const offset = new Date().getTimezoneOffset() * -1;
+  const localDate = addMinutes(date, offset);
+
+  switch (granularity) {
+    case "perMinute":
+      return format(localDate, "HH:mm:ss");
+    case "perHour":
+      return format(localDate, "MMM d, HH:mm");
+    case "perDay":
+      return format(localDate, "MMM d");
+    default:
+      return format(localDate, "Pp");
+  }
+};
+
+export const useFetchTimeseries = (initialTimeseries: LogsTimeseriesDataPoint[]) => {
+  const { searchParams } = useLogSearchParams();
+
+  const filters = useMemo(
```

Comment:
what does the useMemo do here?
isn't it the same as just accessing the values directly?

Reply:
I guess this one is a bit redundant, coz trpc diffs them too. We can omit